analyzers/src/SonarAnalyzer.Common/Rules/StringConcatenationInLoopBase.cs

```diff
@@ -69,19 +68,20 @@ private void CheckCompoundAssignment(SonarSyntaxNodeReportingContext context)
             }
         }
 
-        private SyntaxNode GetInnerMostLeftOfConcatenation(TBinaryExpression binaryExpression)
+        private bool IsIdentifierOnTheRight(SyntaxNode identifier, SyntaxNode expression)
         {
-            var nestedLeft = Language.Syntax.BinaryExpressionLeft(binaryExpression);
-            while (nestedLeft is TBinaryExpression { } nestedBinary)
+            while (expression is TBinaryExpression { } && Language.Syntax.IsAnyKind(expression, ExpressionConcatenationKinds))
             {
-                if (!Language.Syntax.IsAnyKind(nestedBinary, ExpressionConcatenationKinds))
+                var left = Language.Syntax.BinaryExpressionLeft(expression);
+                var right = Language.Syntax.BinaryExpressionRight(expression);
+
+                if (Language.Syntax.AreEquivalent(left, identifier) || Language.Syntax.AreEquivalent(right, identifier))
                 {
-                    return null;
+                    return true;
                 }
-
-                nestedLeft = Language.Syntax.BinaryExpressionLeft(nestedBinary);
+                expression = left;
```

Comment:
A bit more precise is this (whit the while of course):
```suggestion
                return 
                     IsIdentifierOnTheRight(SyntaxNode identifier, left)
                     || IsIdentifierOnTheRight(SyntaxNode identifier, right);
```
But this is only necessary if we also would step into parenthesis or other expressions: `_ = "a" + (s  + "b");`. I don't think we need to support that as the parenthesis are superfluous. Just add comment here:
```cs
// No need to recurse into the right branch as the only useful concatenation is flat `"a" + "b" + s` 
// `"a" + (s  + "b")` seems not worth to support.
```
This also means we have FNs for cases like this:
```cs
s = "a" + (s == null ? "Empty" : s);
```
I leave it to you to decide if you want to add support for this.